aether-odk-module/aether/odk/api/xform_utils.py

```diff
@@ -649,13 +655,15 @@ def __get_xform_instance_skeleton(xml_definition):
     # this contains the data skeleton
     # take all the xpaths and rest of meaningful data from here
     instance = __get_xform_instance(xform_dict, with_root=True)
-    for xpath, has_children in __get_all_paths(instance):
+    for xpath, has_children, visualizer in __get_all_paths(instance):
         schema[xpath] = {
             'xpath': xpath,
             'type': 'group' if has_children else 'string',
             'required': False,
             'label': __get_xform_label(xform_dict, xpath, itexts),
         }
+        if visualizer:
```

Comment:
This is an internal estructure, you don't need to be so cautious, also either to use the same complicated key name ;-)

```python
        schema[xpath] = {
            # ...
            'visualizer': visualizer,
        }
```